lib/types.ts

```diff
@@ -42,9 +44,86 @@ export interface Operation<T> {
   [Symbol.iterator](): Iterator<Instruction, T, any>;
 }
 
+/**
+ * A value that is both an {@link Operation} _and_ `Promise`.
+ *
+ * Futures are operations that are implicitly associated with an Effection scope
+ * and so they can be freely `await`ed within any async functions. However, they
+ * can also be evaluated directly within another operation, so among other
+ * things, if the operation resolves synchronously, it will continue within the
+ * same tick of the run loop.
+ */
 export interface Future<T> extends Promise<T>, Operation<T> {}
 
+/**
+ * A handle to a concurrently running operation that lets you either use the
+ * result of that operation, or shut it down.
+ *
+ * When it is run or spawned, an operation executes concurrently with
+ * the rest of the program. The `Task` is both an {@link Operation} and a
+ * `Promise` that lets you consume the result of that operation.
+ *
+ * @example
+ * ```javascript
+ * import { run, sleep } from "effection";
+ *
+ * let task = run(function*() {
+ *   yield* sleep(100);
+ *   return "hello world"
+ * });
+ *
+ * console.log(await task); //=> "hello world"
+ * ```
+ *
+ * A task can also be created from within an operation by using the
+ * {@link spawn()} operation.
+ *
+ * @example
+ * ```javascript
+ * import { run, spawn, sleep } from "effection";
+ *
+ * await run(function*() {
+ *   let task = yield* spawn(function*() {
```

Comment:
I'm not sure if this is mentioned elsewhere, but it needs to be made very clear that tasks **must** be yielded/awaited-to or else when the parent task completes it will shut down the child task whether the child task is completed or not.

I'm not sure if there's a design principles doc but that should be a bullet point in it.

Reply:
Yeah, it's definitely a subtlety to it. What about this section? https://effection.deno.dev/docs/scope

I think we could add it to that section (which is incomplete) and then link to it from the type docs. Would that work?

Reply:
@neurosnap I added a quick reference to the scoping guide. I think we can update the scoping guide too, but that would be in a later PR.